src/ddmd/func.d

```diff
@@ -5294,9 +5294,18 @@ extern (C++) final class InvariantDeclaration : FuncDeclaration
  */
 extern (C++) static Identifier unitTestId(Loc loc)
 {
+    // there's a counter per module in case there are two unit tests on one line
+    static __gshared int[const(char)*] counterPerModule;
```

Comment:
@klickverbot Perhaps, but that's well beyond the scope of this PR - it's an overhaul, however minor, of the compiler. Also, `__traits(getUnitTests)` is a bit... special. It seems unlikely (to me anyway) that the "resetability" will affect any other features in a similar way. Lambdas aren't referenced by name by other modules.